### DISCUSSION TITLE
Possible to set up custom tunes or tones?

### DISCUSSION THREAD

**Discussion by BirbByte** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#665] [Q&A]
I'm curious to know a bit more about these 'popcorn.mel' and 'furelise.mel' files and how they're created. For fun I want to know if there's any way to have the beeper unit in the display play a different sound, possibly from a MIDI file?

**Comment by visualapproach** [comment]
Yes you can make your own songs. There is no tool for doing so, so you have to use some generalist programs like notepad++ and excel to make it somewhat easy. Actually I asked chatGPT to get the notes and format it as I wanted which it did perfectly. Except that it made the songs up instead of using the songs I asked for. Really annoying:D
I can post a format example here later

**Reply by BirbByte** [reply]
Yeah if you can supply an example that would be great! I’ve gone into Notepad++ and I just don’t understand what all the numbers mean and in this case if I were to ask ChatGPT, I have no idea what I’m asking for or what it’s supposed to be creating

**Comment by visualapproach** [comment]
Ok so here is the format. Every line must end with '\n' aka NewLine character. Using Notepad++ show all symbols and you will be able to confirm this. If it says [CR][LF] it's okay.
I have commented below, but no comments allowed in real file.

300 ;One beat duration in milliseconds. Set the pace of the song.
0.9 ;Duty cycle. If set to 1.0 there will be no break between notes.
659 ;Frequency in Hz
0.5 ; note duration, fraction of beat duration. (Beat duration * this value will be note duration in ms)
622 ;freq
0.5 ;duration. This is a half note
659 ;freq
0.5 ;duration
622
0.5
659
0.5
494
0.5
587
0.5
523
0.5
440
0.5
0 ;Zero Hz = paus
0.5 ;half note
repeat until song end

Save with .mel extension and upload to ESP with the upload menu in the Web UI

**Reply by BirbByte** [reply]
Perfect. I’ll have a play around with that later thanks 😀
